lib/algos/list/delimiterNonCurrent.js

```diff
@@ -148,18 +150,25 @@ class DelimiterNonCurrent extends Delimiter {
             return FILTER_ACCEPT;
         }
 
-        // The following version is pushed only if the "stale date" (picked up from the previous version)
-        // is available (JSON.parse has not failed) and, if the "beforeDate" argument is specified,
-        // the "stale date" is older than the "beforeDate".
+        // The following version is pushed only:
```

Comment:
If I understand the code flow, it seems that we always have to parse the metadata in any case. I'd suggest to parse it at the beginning then, and use the fields directly thereafter, as it is now the tentative optimizations only seem to clutter the code.